src/lib.rs

```diff
@@ -1,18 +1,23 @@
 extern crate rustc_serialize;
+use std::string;
 
 pub enum Value {
   Boolean(bool),
   TinyInt(u8),
+  TinyText(Result<String, string::FromUtf8Error>)
 }
 
-pub fn unpack(bytes: Vec<u8>) -> Option<Value> {
-  let header_byte = bytes[0];
+pub fn unpack(mut bytes: Vec<u8>) -> Option<Value> {
+  let header_byte = bytes.remove(0);
```

Comment:
My thought was that we would just read the data without mutating it.  I think the one thing that you should keep in mind that the tests aren't covering is that a PackStream stream can have multiple values, so it's important to take the header byte and then only pull out the bytes which are specified by that header byte.  So if the header byte is `0x82` you should pull out just the next two bytes.  I think that this is possible in rust with brackets like `bytes[1..3]`